### PR TITLE
Fix/sshd config

### PR DESCRIPTION
**Pull request by OliverO2**
##### Pull Request Details:

* Type: **Bug Fix**

* Impact: **Normal**

* Reference to related issue (URL): #2362

* How was this pull request tested? On Ubuntu 18.04.4 LTS

* Brief description of the changes in this pull request: See issue.

### CODE REVIEW

File: usr/share/rear/build/default/500_ssh_setup.sh [hunk 1]
```diff
@@ -8,31 +8,62 @@ has_binary ssh || has_binary sshd || return 0
...
-    fi
+if [[ -f "$sshd_config_file" ]]; then
+    # Enable root login with a password only if SSH_ROOT_PASSWORD is set
+    local password_authentication_value=no
+    [[ -n "$SSH_ROOT_PASSWORD" ]] && password_authentication_value=yes
+
+    # List of setting overrides required for the rescue system's sshd - see sshd_config(5)
+    # Each list element must be a string of the form 'keyword [value]' or a comment '#...'.
+    # If value is missing, the respective keyword will effectively be set to its default value.
+    sshd_setting_overrides=(
```
Comment (jsmeix) [inline usr/share/rear/build/default/500_ssh_setup.sh:22]: I think this should be `local sshd_setting_overrides=(`

File: usr/share/rear/build/default/500_ssh_setup.sh [hunk 2]
```diff
@@ -8,31 +8,62 @@ has_binary ssh || has_binary sshd || return 0
...
+        "ListenAddress"
+        # Use default handling of idle messages
+        "ClientAliveInterval"
+        # End comment
+        "### END ReaR overrides"
+    )
+
+    # Create sed options containing a list of commands to patch the existing sshd configuration file.
+    local sed_patch_options=()
+    for sshd_option in "${sshd_setting_overrides[@]}"; do
```
Comment (jsmeix) [inline usr/share/rear/build/default/500_ssh_setup.sh:45]: I think there should be also `local keyword value`

**Comment by jsmeix** [comment]
@rear/contributors 
when there are no objections I would like to merge it today evening.

**Comment by OliverO2** [comment]
@gdha @jsmeix 
Thanks to both of you for reviewing. Always a pleasure!

File: usr/share/rear/build/default/500_ssh_setup.sh (hunk 1 above)
Comment (OliverO2) [reply usr/share/rear/build/default/500_ssh_setup.sh]: Done.

File: usr/share/rear/build/default/500_ssh_setup.sh (hunk 2 above)
Comment (OliverO2) [reply usr/share/rear/build/default/500_ssh_setup.sh:46]: Done.

**Pull request merged**